hw/ip/i2c/doc/_index.md

```diff
@@ -1,3 +1,223 @@
+- The periods between transmitted bytes when the state-machine should stop
+transmission and instead read back a fixed number of bytes.
+- Which bytes should be preceded by a START symbol.
+- Which bytes should be followed by a STOP symbol
+The format indicator consists of 13-bits. That is of one single Format Byte (entered into the format FIFO through {{< regref FDATA.FBYTE >}}), and five (5) 1-bit flags (entered into the format FIFO through registers {{< regref FDATA.READ >}}, {{< regref FDATA.RCONT >}}, {{< regref FDATA.START >}}, {{< regref FDATA.STOP >}} and {{< regref FDATA.NAKOK >}})
+
+The I2C reads each format indicator from the head of FMT_FIFO, and processes them in turn.
+If none of the flags are set for the format format indicator none, the I2C FSM simply transmits the Format Byte onto the SCL and SDA pins according to the specification, waits for acknowledgement, and then proceeds to the next format indicator.
+The format flags modulate the behavior as follows.
+- READ (Corresponds to {{< regref FDATA.READ >}}):
+Signifies the Format Byte ({{< regref FDATA.FBYTE >}}) should be treated as an unsigned number, R, and prompts the state machine to read R bytes from the target device.
+Bytes read from the bus, are inserted into the RX FIFO where they can be accessed by software.
+A value of 0 is treated as a read of 256B.
+To read a larger byte stream, multiple 256B reads can be chained together using the RCONT flag.
+- RCONT (Corresponds to FIFO inputs {{< regref FDATA.RCONT >}}, Only used with READ):
+If RCONT is set, the Format Byte represents part of a longer sequence of reads, allowing for reads to be chained indefinitely.
+The RCONT flag indicates the the final byte returned with the current read
+should be responded to which an ACK, allowing the target to continue sending data.
+(Note that the first R-1 bytes read will still be acknowledged regardless
+of whether RCONT is asserted or not).
+- START (Corresponds to {{< regref FDATA.START >}}, Ignored when used with READ):
+Issue a START condition before transmitting the Format Byte on the bus.
+This flag may also be used to issue a repeated start condition.
+- STOP (Corresponds to {{< regref FDATA.STOP >}}):
+Issue a STOP signal after processing this current entry in the FMT FIFO.
+Note that this flag is not compatible with (READ & RCONT), and will cause bus conflicts.
+- NAKOK (Corresponds to {{< regref FDATA.NAKOK >}}, Not compatible with READ):
+Typically every byte transmitted must also receive an ACK signal, and the IP will raise an exception if no ACK is received.
+However, there are some I2C commands which do not require an ACK.
+In those cases this flag should be asserted with FBYTE indicating no ACK is expected and no interrupt should be raised if the ACK is not received. 
+
+### Timing Control Registers
+
+For Standard-mode, Fast-mode and Fast-mode Plus, the timing requirements for each transaction are detailed in Table 10 of the I2C specification.
```

Comment:
style nit: it's called "standard mode" in the Features section at the top. Please choose one spelling.

"I2C specification" could be linked to the spec again.